torch/_decomp/decompositions.py

```diff
@@ -1963,6 +1963,12 @@ def uniform(
     )
 
 
+@register_decomposition(aten.exponential_)
```

Comment:
what about casting halfs to higher precision for intermediate computations?

Reply:
Decorating it with `pw_cast_for_opmath` should be good enough right?

Reply:
Oh thanks for the catch, `pw_cast_for_opmath` should do. I'm checking if other `ELEMENTWISE_TYPE_PROMOTION_KIND` than the default would be applicable.